### PR TITLE
Update OutlineButton for latest Material spec, fix flicker issue, support materialTapTargetSize

### LINKED ISSUE #22755: Dark Material Theme: OutlineButton is flickering when pressed, showing black opaque rectangle
Flutter version: Latest master

The `OutlineButton` is displaying a black rectangle shortly after being pressed, and flickers on release.

When I set a button `color`, the black rectangle no longer appears, instead the opaque version of the color is used as a background while the button is pressed. Transparency values are ignored. It all seems broken.

Other buttons like `FlatButton` do not show the same behavior.

ezgif-3-af8f818f2ae8

```dart
import 'package:flutter/material.dart';

void main() => runApp(new MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return new MaterialApp(
      title: 'Stroke Button Test',
      theme: ThemeData.dark(),
      home: new HomePage(),
    );
  }
}

class HomePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text('Outline Button Test')),
      body: Center(
        child: OutlineButton(
          onPressed: () {},
          child: Text('Do something'),
        ),
      ),
    );
  }
}
```

### PR DESCRIPTION
**Pull request by sir-boformer**
fixes #22755 and #22778

* no elevation on press (base on `FlatButton` instead of `RaisedButton`). This also fixes #22755 because there is no longer a need to fill the button with an opaque color on press, and makes it possible to use the button on colored/image backgrounds.
* 1px border instead of 2px
* Transparent black border (white for dark theme) which does not change on press
* If no `highlightedBorderColor` is defined, use normal border color while pressed.
* materialTapTargetSize

I used `0x3b000000` (`0x3bffffff` for dark theme) as the default border color (if placed on white background). That's what is used as a color in the material spec images, although in the "Specs" section states that the border color should be `0x1e000000`, which is much brighter.

example app

Spec for comparison:

### CODE REVIEW

**Comment by sir-boformer** [comment]
@HansMuller This is ready for review

**Comment by HansMuller** [comment]
Although I'm disappointed to see the tricky [transparent - to opaque - to elevated] transition go, the spec no longer includes those things.  So out with the old.

Will give it a review soon.

**Comment by willlarche** [comment]
Please hold on this PR while I get clarifications from Material designers around some of the values.

**Comment by willlarche** [comment]
Yeah everything about the states I need clarifications on. Also, whether it's supposed to have a transparent background or onSurface.

**Comment by sir-boformer** [comment]
Got a reply from the designers?

**Comment by willlarche** [comment]
Thanks for your patience. Yes, I did!

- Fill color: ColorScheme.Surface (0% opacity) (kind of a funny value but ok!)
- Stroke color: #000000  (12% opacity)
- Hover fill color: #3B00DD (4% opacity)
- Focused fill color: #3B00DD (12% opacity)
- Ink color: #3B00DD (16% opacity)

**Comment by sir-boformer** [comment]
Are those values for the ink response (focused fill color and ink color) also used for other buttons, e.g. the flat button? In that case I could just update the `ButtonTheme` defaults.

Right now `ButtonTheme`  defines these defaults:

* Focused fill color: 16% opacity
* Ink color: 12% opacity

File: packages/flutter/lib/src/material/outline_button.dart [hunk 1]
```diff
@@ -63,39 +47,37 @@ class OutlineButton extends MaterialButton {
...
     Color highlightColor,
     Color splashColor,
-    double highlightElevation,
     this.borderSide,
     this.disabledBorderColor,
     this.highlightedBorderColor,
     EdgeInsetsGeometry padding,
     ShapeBorder shape,
     Clip clipBehavior = Clip.none,
+    MaterialTapTargetSize materialTapTargetSize,
```
Comment (jonahwilliams) [inline packages/flutter/lib/src/material/outline_button.dart]: The materialTapTargetSize is already provided by the theme. This has some fairly oblique functionality and I don't think it makes sense to customize at the level of an individual button.

Unless of course it somehow isn't retrieving the theme value, then we should fix that instead.

Comment (sir-boformer) [reply packages/flutter/lib/src/material/outline_button.dart]: `RaisedButton`, `FlatButton`, `FloatingActionButton` and `MaterialButton` all have a `materialTapTargetSize` property. So I think it makes sense that `OutlineButton` gets one as well.

Comment (jonahwilliams) [reply packages/flutter/lib/src/material/outline_button.dart]: ahh you'e right (I wonder when that changed..), carry on!

**Comment by willlarche** [comment]
@sir-boformer , open an issue to investigate that. No need to block your work here. There may be issues with it we have to run my the TL (changes are like that.)

**Comment by Borduni** [comment]
Is this ready to get merged? Facing that issue too.

**Comment by jonahwilliams** [comment]
+ @HansMuller could you take a look at this PR when you have the chance?

**Comment by dariotrombello** [comment]
Any movement here?

**Comment by HansMuller** [comment]
I'm going to try and make some progress on actually landing most of this PR next week. It's been a long time since the changes were first proposed (that's my fault, sorry about that) so I'll probably create a new PR.

**Comment by dariotrombello** [comment]
Thank you! No problem. I'm anyway happy that this is fixed soon because I use many dark OutlineButtons in my app and because of this flickering issue i needed to set a brighter colour when tapping on it (splash colour) which looks pretty ugly.

**Comment by Ekkarat304** [comment]
Thank you

**Comment by HansMuller** [comment]
I'm going to close this because the changes have landed as part of #27575.

Thanks to @sir-boformer  for the contribution and thank you-all for making sure this work wasn't overlooked!

**Pull request closed without merging**